src/ir_passes/mono/utils/comp.rs

```diff
@@ -10,6 +10,9 @@ use super::{Base, IntoBase, Underlying};
 pub struct UnderlyingComp<'a>(&'a ir::Component);
 
 impl<'a> UnderlyingComp<'a> {
+    pub fn comp(&self) -> &'a ir::Component {
```

Comment:
We want to avoid exposing comp on underlying and instead expose methods that force the user of the struct to wrap things in Underlying.

Reply:
The reason for doing this was because `relevant_vars` can't be run on an underlying component easily.

Reply:
I guess I can rewrite it to use `IdxLike`?

Reply:
Yeah, probably!

Reply:
Or just define a wrapper method that unwraps the underlying comp and the underlying index